librosa/core/time_frequency.py

```diff
@@ -1682,3 +1713,222 @@ def samples_like(X, hop_length=512, n_fft=None, axis=-1):
+    if not match:
+        raise ParameterError('Improper key format: {:s}'.format(key))
+
+    pitch_map = {'C': 0, 'D': 2, 'E': 4, 'F': 5, 'G': 7, 'A': 9, 'B': 11}
+    acc_map = {'#': 1, '': 0, 'b': -1, '!': -1, '♯': 1, '♭': -1}
+
+    tonic = match.group('tonic').upper()
+    accidental = match.group('accidental')
+    offset = acc_map[accidental]
+
+    scale = match.group('scale')[:3].lower()
+
+    # Determine major or minor
+    major = (scale == 'maj')
+
+    # calculate how many clockwise steps we are on CoF (== # sharps)
+    if major:
+        tonic_number = ((pitch_map[tonic] + offset) * 7) % 12
+    else:
+        tonic_number = ((pitch_map[tonic] + offset) * 7 + 9) % 12
+
+    # Decide if using flats or sharps
+    # Logic here is as follows:
+    #   1. respect the given notation for the tonic.
+    #      Sharp tonics will always use sharps, likewise flats.
+    #   2. If no accidental in the tonic, try to minimize accidentals.
+    #   3. If there's a tie for accidentals, use sharp for major and flat for minor.
+
+    if offset < 0:
+        # use flats explicitly
+        use_sharps = False
+
+    elif offset > 0:
+        # use sharps explicitly
+        use_sharps = True
+
+    elif 0 < tonic_number < 6:
+        use_sharps = True
+
+    elif tonic_number > 6:
+        use_sharps = False
+
+    else:
+        # Equal numbers of sharps and flats for tonics 0 and 6
+        # break ties for major => sharp, minor => flat
+        use_sharps = major
+
+    # Basic note sequences for simple keys
+    notes_sharp = ['C', 'C♯', 'D', 'D♯', 'E', 'F', 'F♯', 'G', 'G♯', 'A', 'A♯', 'B']
+    notes_flat = ['C', 'D♭', 'D', 'E♭', 'E', 'F', 'G♭', 'G', 'A♭', 'A', 'B♭', 'B']
+
+    # These apply when we have >= 6 sharps
+    sharp_corrections = [(5, 'E♯'), (0, 'B♯'), (7, 'F𝄪'),
+                         (2, 'C𝄪'), (9, 'G𝄪'), (4, 'D𝄪'), (11, 'A𝄪')]
+
+    # These apply when we have >= 6 flats
+    flat_corrections = [(11, 'C♭'), (4, 'F♭'), (9, 'B𝄫'),
+                        (2, 'E𝄫'), (7, 'A𝄫'), (0, 'D𝄫')]  # last would be (5, 'G𝄫')
+
+    # Apply a mod-12 correction to distinguish B#:maj from C:maj
+    n_sharps = tonic_number
+    if tonic_number == 0 and tonic == 'B':
+        n_sharps = 12
+
+    if use_sharps:
+        # This will only execute if n_sharps >= 6
+        for n in range(0, n_sharps - 6 + 1):
+            index, name = sharp_corrections[n]
+            notes_sharp[index] = name
+
+        notes = notes_sharp
+    else:
+        n_flats = (12 - tonic_number) % 12
+
+        # This will only execute if tonic_number <= 6
+        for n in range(0, n_flats - 6 + 1):
+            index, name = flat_corrections[n]
+            notes_flat[index] = name
+
+        notes = notes_flat
+
+    # Finally, apply any unicode down-translation if necessary
+    if not unicode:
+        translations = str.maketrans({'♯': '#', '𝄪': '##', '♭': 'b', '𝄫': 'bb'})
+        notes = list([n.translate(translations) for n in notes])
```

Comment:
silly question, but do we really need the brackets around the list comprehension syntax? would

`list(n.translate(translations) for n in notes)`

suffice?